src/rgw/rgw_user.cc

```diff
@@ -2462,8 +2465,11 @@ int RGWUserAdminOp_Subuser::remove(RGWRados *store, RGWUserAdminOpState& op_stat
 
 
   ret = user.subusers.remove(op_state, NULL);
-  if (ret < 0)
+  if (ret < 0) {
+    if (ret == -ENOENT)
+      ret = -ERR_NO_SUCH_SUBUSER;
```

Comment:
With this logic, do you think that `ENOENT` will be returned at another place in the same originating function or by a different function in the future, so that the execution may reach this point ? Instead it's safe returning a specific error such as `ERR_NO_SUCH_SUBUSER' right ?. What do you say ?